### PR TITLE
fix: flakeMetadata and locked flakeRef types

### PR DESCRIPTION
**Pull request by ysndr**
* add some missing fields in the attributes of git flake refs
* fix the rendering of attributes to query params (for flakerefs that contain bools)
* add a ci action to continuously test

### CODE REVIEW

File: crates/runix/src/flake_ref/git_service.rs [hunk 1]
```diff
@@ -38,10 +38,10 @@ pub struct GitServiceAttributes {
 
     pub rev: Option<Rev>,
 
-    #[serde(flatten)]
+    #[serde(rename = "narHash")]
     pub nar_hash: Option<NarHash>,
 
-    #[serde(flatten)]
+    #[serde(rename = "lastModified")]
```
Comment (mkenigs) [inline crates/runix/src/flake_ref/git_service.rs:44]: Looks like we don't have test coverage for either of these?

File: crates/runix/src/flake_ref/git_service.rs [hunk 2]
```diff
@@ -38,10 +38,10 @@ pub struct GitServiceAttributes {
 
     pub rev: Option<Rev>,
 
-    #[serde(flatten)]
```
Comment (mkenigs) [inline crates/runix/src/flake_ref/git_service.rs:41]: Optional: `#[serde(rename_all = "camelCase")]`

File: crates/runix/src/flake_ref/mod.rs [hunk 3]
```diff
@@ -395,6 +367,8 @@ pub(super) mod tests {
     use std::fmt::Debug;
     use std::fs::{self, File};
 
+    use serde_json::json;
+
```
Comment (mkenigs) [inline crates/runix/src/flake_ref/mod.rs]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    - name: check clippy
+      run: flox activate -e .#runix -- cargo clippy
+
+    - name: check format
+      run: |
+        # flox activate does not activate with the current flake's self context :X
+        flox build .#rustfmt
+        export PATH="./result/bin:$PATH"
+        flox activate -e .#runix -- cargo fmt --check
```
Comment (mkenigs) [inline .github/workflows/ci.yml:35]:
```suggestion
        flox develop .#runix --command cargo fmt --check
```

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (ysndr) [reply .github/workflows/ci.yml:35]: Flox develop only works in interactive environments.. for some reason we abort in non interactive cases with an error message about that - considering to open a flox bug report for that.

Comment (mkenigs) [reply .github/workflows/ci.yml:35]: I believe `flox print-dev-env` should work then

Comment (mkenigs) [reply .github/workflows/ci.yml:35]: Maybe we should just tweak the error message to say to use `print-dev-env`

Comment (mkenigs) [reply .github/workflows/ci.yml:35]: Oh no you're right it's a bug that we error when `--command` is passed

**Review by mkenigs — APPROVED** [review]
Maybe open an issue about `--command` and link that issue in `ci.yml`?

**Pull request merged**